src/client.ts

```diff
@@ -575,15 +585,30 @@ export class Client {
 		const { limit = Infinity, ...actualParams } = params;
 		const resolvedParams = { pageSize: MAX_PAGE_SIZE, ...actualParams };
 
-		const result = await this.get<TDocument>(resolvedParams);
+		const documents: TDocument[] = [];
```

Comment:
I think we could simplify the code with the following function. The main idea is to use a generator:

```js
const browseAPI = async function*(url) {
  let nextUrl = url;

  while(nextUrl) {
    const response = await fetchPage(nextUrl);

    const { 'next_page': nextPage } = response;

    nextUrl = nextPage;

    yield response;
  }
};
```

The `getAll` function could rely on this and add a waiting time between each call.

Reply:
Maybe I'm missing something, but I don't think a generator function adds anything here since can inline that logic into the method (i.e. no need for an abstraction). I'll update the code to use a static delay between requests which will result in simpler code.

Reply:
The full looping logic is now as follows:

```ts
const documents: TDocument[] = [];
let latestResult: prismicT.Query<TDocument> | undefined;

while (
	(!latestResult || latestResult.next_page) &&
	documents.length < limit
) {
	const page = latestResult ? latestResult.page + 1 : undefined;

	latestResult = await this.get<TDocument>({ ...resolvedParams, page });
	documents.push(...latestResult.results);

	if (latestResult.next_page) {
		await new Promise((res) => setTimeout(res, GET_ALL_QUERY_DELAY));
	}
}

return documents.slice(0, limit);
```

This retains use of `this.get` as a fetching and typing abstraction.